config.json

```diff
@@ -168,28 +161,37 @@
       "defaultValues":{
         "toolName": "msbuild",
         "settings": {
-          "MsBuildParameters":"default",
-          "MsBuildLogging":"default"
+          "MsBuildLogging":"/flp:v=normal;LogFile=clean.log"
         }
       }
     }
   },
   "tools": {
     "msbuild": {
-      "run": {
-        "windows": "Tools/msbuild.cmd",
-        "unix": "Tools/msbuild.sh"
+      "osSpecific":{
+        "windows": {
+          "defaultParameters": "/nologo /verbosity:minimal /clp:Summary /maxcpucount /nodeReuse:false /l:BinClashLogger,Tools\\net45\\Microsoft.DotNet.Build.Tasks.dll;LogFile=binclash.log",
+          "path": "Tools/msbuild.cmd"
+        },
+        "unix":{
+          "defaultParameters": "/nologo /verbosity:minimal /clp:Summary /maxcpucount /l:BinClashLogger,Tools/Microsoft.DotNet.Build.Tasks.dll;LogFile=binclash.log",
+          "path": "Tools/msbuild.sh"
+        }
       },
       "valueTypes": {
         "property": "/p:{name}={value}",
         "target": "/t:{name}",
         "internal": "/{name}"
       }
     },
-    "terminal": {
-      "run": {
-        "windows": "cmd",
-        "unix": "sh"
+    "script": {
+      "osSpecific":{
+        "windows": {
+          "filesExtension": "cmd"
+        },
```

Comment:
The way the tool `script` works is that it attaches the fileExtension to the name of a file that comes from the setting `Project`.
Before we had it like [this](https://github.com/dotnet/buildtools/pull/976/files#diff-06b2d3b23dce96e1619d2b53d6c947ecL190) and the logic to attached the extension was hard coded in the code when the toolname was `terminal`.

With this new value I delete the hard coded things in run.exe and I added a more clear name for the file extension